instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -277,3 +292,27 @@ def _trace_retry(*args, **kwargs):
         # Use `str(reason)` instead of `reason.message` in case we get
         # something that isn't an `Exception`
         span.set_attribute(_TASK_RETRY_REASON_KEY, str(reason))
+
+    def update_task_start_time(self, task_id):
+        cur_time = default_timer()
+        start_time = (
+            cur_time - self.task_id_to_start_time[task_id]
+            if task_id in self.task_id_to_start_time
+            else cur_time
+        )
+        self.task_id_to_start_time[task_id] = start_time
```

Comment:
I still can't follow this logic. Are you trying to record the task duration? Isn't this `cur_time - self.task_id_to_start_time[task_id]` an elapsed time?

Reply:
No, it's the task start time.

Reply:
How is it a start time when it has this line `cur_time - self.task_id_to_start_time[task_id]`? You appear to want to record the task duration for a histogram? Or do you want to record the start time of the task as a histogram, and what does it mean to record the start time of the task as a histogram?

Reply:
Ok I understand why you try to tell me, I changed the name that it will be clearer.
This variable is used to count the duration time, but in the first time we call it the task id still isn't exist in task_id_to_start_time so we save the current time and in the next time we call it we calculate the duration that is the current time minus the time we saved in task_id_to_start_time in the first call.